components/api/core/utils.bal

```diff
@@ -39,3 +40,120 @@ function updatePayloadWithUserInfo (json payload, string field) returns error? {
+
+                    if (registryScopeForDeleteManifest is string) {
+                        log:printDebug(io:sprintf("Registry scope for deleting manifest from registry api: %s", registryScopeForDeleteManifest));
+
+                        string | error? tokenToDeleteManifest = docker_registry:getTokenFromDockerAuth(userId, token, registryScopeForDeleteManifest);
+
+                        if (tokenToDeleteManifest is string) {
+                            log:printDebug("Retrived a token from docker auth to invoke deleteManifest end point");   
+
+                            error? artifactDeleteResult = docker_registry:deleteManifest(orgName, imageName, manifestDigest, tokenToDeleteManifest);
+
+                            if (artifactDeleteResult is error) {
+                                return artifactDeleteResult;
+                            } else {
+                                log:printDebug(io:sprintf("Artifact \'%s/%s:%s\' is successfully deleted from the registry", orgName, imageName, artifactVersion));
+                            }
+                        } else {
+                            return tokenToDeleteManifest;
+                        }                        
+                    } else {
+                        return registryScopeForDeleteManifest;
+                    }                
+                } else {
+                    return manifestDigest;
+                }
+            } else {
+                return tokenToGetManifestDigest;
+            }            
+        } else {
+            return registryScopeForGetManifest;
+        }
+    } else {
+        return token;
+    }
+}
+
+public function deleteImageFromResitry(string orgName, string imageName) returns error? {
+    string imageDirectoryPath = io:sprintf("%s/%s/%s", constants:DOCKER_REGISTRY_REPOSITORIES_FILEPATH, orgName, imageName);
+    log:printInfo(io:sprintf("Deleting the image \'%s/%s\' from the registry. Image directory : %s", orgName, imageName, imageDirectoryPath));
+    internal:Path directoryToBeDeleted = new (imageDirectoryPath);
+
+    if (directoryToBeDeleted.isDirectory()) {
+        var deleteResult = directoryToBeDeleted.delete();
+
+        if (deleteResult is error) {
+            error er = error("Unexpected error while deleting the image from the registry");
+            return er;
+        } else {
+            log:printInfo(io:sprintf("Image \'%s/%s\' is successfully deleted from the registry", orgName, imageName));
+        }
+    } else {
+        error er = error(io:sprintf("Image directory \'%s\' is not found in the registry", imageDirectoryPath));
+        return er;
+    }
+}
+
+public function deleteOrganizationFromResitry(string orgName) returns error? {
+    string orgDirectoryPath = io:sprintf("%s/%s", constants:DOCKER_REGISTRY_REPOSITORIES_FILEPATH, orgName);
```

Comment:
Let's check for empty values. In a case if we call with an empty orgname while reusing this method, it will wipeout the whole docker registry